shared/src/main/scala/scodec/codecs/VarPackedDecimalCodec.scala

```diff
@@ -0,0 +1,34 @@
+package scodec
+package codecs
+
+import java.nio.ByteBuffer
+
+import scodec.bits.BitVector
+
+import scala.annotation.tailrec
+
+
+private [codecs] final class VarPackedDecimalCodec extends Codec[Long] {
+  override def sizeBound = SizeBound.bounded(1L, 9L)
+
+  def encode (i: Long): Attempt[BitVector] = {
+    require(i >= 0, "VarPackedBcd cannot encode negative longs")
+    runEncoding(i, Attempt.successful(BitVector.empty))
+  }
+
+  def decode (buffer: BitVector): Attempt[DecodeResult[Long]] = runDecoding(buffer, 0L)
+  
+  @tailrec
+  private def runDecoding (buffer: BitVector, value: Long): Attempt[DecodeResult[Long]] = {
+    if (buffer.size == 0) Attempt.successful(DecodeResult(value, buffer))
```

Comment:
Can you change `buffer.size` to `buffer.isEmpty`? If the `BitVector` has suspend nodes, calling `.size` will force them all to be computed.